### ISSUE TITLE
Refactor 'onboarding status' data

### ISSUE THREAD

**Issue by IagoSRL** [loconomics/loconomics#396]
### Preface
Working at #374, I implemented a workaround that makes the app remember the job title selected/generated by a user at sign-up for the onboarding steps that require to select one.

**How it works:** the jobTitleID is saved in local storage as `onboardingJobTitleID` each time a value is set at `app.model.onboarding.selectedJobTitleID` and is read at app start-up to initialize that variable, allowing the user to resume the onboarding process. Is never saved at database, in the REST API only exists as part of the response of sign-up.

**Progress of onboarding:** The other data we have that allows user to start/resume the onboarding process is the `onboardingStep`; this field exist at the [users] table, is provided by the *profile* REST API and returned by the response of the sign-up and log-in; as part of that, is saved in local storage under the profile data.
In order to fix #380 #381, the locally stored profile data is read at app start-up to initialize the app.model.onboarding, allowing the user to start or resume the onboarding process.

### Goal
It's a bit confusing in the code to have that data saved and read in different places and APIs. Both fields are part of the 'onboarding status', so is better to manage unified.
And there is still an edge case where the bug #374 reproduces again (explained later).

### Refactor Proposal
- Create a table named [UserOnboardingStatus] (or just [UserOnboarding]) where store and track the progress/step and jobTitleID.
  - It moves the field `onboardingStep` from [users], where is removed; prior removal, the data is copied for every user record with a value on that field by the upgrade script.
- Remove from LcRest.UserProfile the field `onboardingStep`.
  - This simplifies too some logic at /api/auth/signup where we have now two response entries that holds that data.
  - The onboarding progress makes no part of the user profile, so makes sense to remove that.
- The onboarding status is returned by signup and login REST APIs similar as now they are returning the `onboardingStep` but including the `jobTitleID` too.
  - Just when the onboarding is in progress, null data otherwise.
  - Because the onboarding jobTitleID is returned by the login too, the behavior is consistent with what happens at signup; right now, a user that continues onboarding in another device, or after logout-login again, the bug #374 will happens again (it's and edge case).
- The REST API to update the onboardingStep must be updated with the jobTitleID (maybe renamed, right now is `PUT /profile/tracking`)
- The internal code to set the onboardingStep must be updated.
- In the App, rather than store onboardingStep and jobTitleID in different places, will be at a new `onboardingStatus` store as part of the AppModel.onboarding API. It simplifies current start-up logic reading the values, and post-sign-up/login saving the values.

### Additional tracking
Since we have a new table, and we already have a REST API that is called on each change of progress in the app to track the current onboarding step, we can add additional fields, like
- CreatedDate DateTime as usual; it does not means the onboarding is started, just user signed-up and record created.
- UpdatedDate TimeTime as usual, time latest change.
- Started DateTime, when user user clicks on the button we have at the 'welcome' step (the user actively starts the process).
- Ended DateTime, to track completion of the onboarding.*

**Important:* this implies we will keep the database record for onboarding progress even when it finishes and we set to null the step, to allow us know the time to complete onboarding.

**Comment by situatedbit** [comment]
Yes! I've been hoping we could do this.

This is a separate conversation, but it will become relevant when we start changing the onboarding process: I want to make individual onboarding steps into their own activities. For example, instead of using ``activities/schedulingPreferences`` for both onboarding and normal use, I want ``activities/onboardingSchedulingPreferences`` _and_ ``activities/schedulingPreferences`` that share common components.

**Comment by IagoSRL** [comment]
@situatedbit Yes, I think that must be the new approach, as commented previously, moving most of the activities code into components.
We can think even if makes sense to have one 'onboarding' activity for all steps, with URLs like 'onboarding/schedulingPreferences', and the view just switches visualization of each component step.
